connector/src/main/java/org/jboss/as/connector/subsystems/datasources/AbstractDataSourceService.java

```diff
@@ -355,12 +355,11 @@ public CommonDeployment deploy(ServiceContainer serviceContainer) throws DeployE
                             throw ConnectorLogger.ROOT_LOGGER.cannotDeploy(ConnectorLogger.ROOT_LOGGER.failedToLoadDataSourceClass(dsClsName, e));
                         }
                     }
-
                     String driverName = dataSourceConfig.getDriver();
                     InstalledDriver installedDriver = driverRegistry.getValue().getInstalledDriver(driverName);
                     if (installedDriver != null) {
-                        String moduleName = installedDriver.getModuleName() != null ? installedDriver.getModuleName().getName()
-                                : null;
+                        // Directly use installedDriver.getModuleName() because it's already a String
+                        String moduleName = installedDriver.getModuleName(); // No need for .getName() here
```

Comment:
The "// No need for ..." comment is going to be confusing once this is merged, as it explains a change you're making, not the actual code that people will be reading. There is no String.getName() method.